rest/replicatortest/replicator_test.go

```diff
@@ -7949,8 +7949,8 @@ func TestReplicatorCheckpointOnStop(t *testing.T) {
 	defer teardown()
 	activeCtx := activeRT.Context()
 
-	// Disable checkpointing at an interval
-	activeRT.GetDatabase().SGReplicateMgr.CheckpointInterval = 0
+	// reduce checkpointing interval temporarily to ensure there is a checkpoint to assert against later in test
+	defer reduceTestCheckpointInterval(50 * time.Millisecond)()
```

Comment:
If we're checkpointing at any interval, how can we be sure the replicator syncronously checkpointed on stop as opposed to the checkpointer ticker doing it?

The intention of the test is that the replicator doesn't rely on the checkpointer interval when a replicator is stopped.

Could we set the interval to something really high if `0` doesn't work?

Reply:
Ah I got the wrong end of the stick. Added your suggestion!

But I wonder if there is contention between the stop checkpointer and the checkpoint interval and thats the reason the checkpoint is missing sometimes? Its hard to fully debug without reproing it. I don't see wither of these log lines in the lastest failiure case on jenkins -> https://github.com/couchbase/sync_gateway/blob/6bf619004c1098d8a69d31fe2e7cda45d108ce6f/db/active_replicator_checkpointer.go#L245C12-L245C12 https://github.com/couchbase/sync_gateway/blob/6bf619004c1098d8a69d31fe2e7cda45d108ce6f/db/active_replicator_checkpointer.go#L248 I did think about adding a 'wait for stat' on the set checkpoint count on the active replicator before trying to assert against it but I don't think that'll help this test case.